core/src/main/java/com/datastax/oss/driver/internal/core/resolver/AbstractResolverFactory.java

```diff
@@ -0,0 +1,7 @@
+package com.datastax.oss.driver.internal.core.resolver;
+
+public abstract class AbstractResolverFactory {
```

Comment:
I am pretty weak in java, but my understanding is that we don't need restrictions that comes from `abstract class` and we can convert it to an interface, could you please double check if it is the case.